clarity/src/vm/contexts.rs

```diff
@@ -219,6 +222,127 @@ pub struct GlobalContext<'a> {
     pub execution_time_tracker: ExecutionTimeTracker,
     #[cfg(feature = "clarity-wasm")]
     pub engine: Engine,
+    pub analysis_db: MemoryAnalysisDatabase,
+}
+
+#[self_referencing]
+pub struct MemoryAnalysisDatabase {
+    mem_store: MemoryBackingStore,
+    #[not_covariant]
+    #[borrows(mut mem_store)]
+    analysis_db: AnalysisDatabase<'this>,
+}
```

Comment:
Why do we need a self-referencing struct here, instead of having the same solution as the entire codebase: an inner field with a reference to something that is declared outside of the struct?